sympy/discrete/recurrences.py

```diff
@@ -111,11 +111,28 @@ def linrec(coeffs, init, n):
 
     if n < k:
         return b[n]
-    terms = [u*v for u, v in zip(_final_coeffs(n), b)]
+    terms = [u*v for u, v in zip(linrec_coeffs(n), b)]
     return sum(terms[:-1], terms[-1])
 
 
-def _linrec_coeffs(c, n):
+def linrec_coeffs(c, n):
+    """
+    It computes the remainder of `x^n` when divided by the polynomial
+    whose coefficients is c.
```

Comment:
Perhaps something like:

    Compute the coefficients of n'th term in linear recursion sequence defined by c.